sqlglot/dialects/tsql.py

```diff
@@ -507,7 +507,7 @@ class Tokenizer(tokens.Tokenizer):
             "PROC": TokenType.PROCEDURE,
             "REAL": TokenType.FLOAT,
             "ROWVERSION": TokenType.ROWVERSION,
-            "SMALLDATETIME": TokenType.DATETIME,
+            "SMALLDATETIME": TokenType.DATETIME2,
```

Comment:
This mapping also looks incorrect, right? `SMALLDATETIME` and `DATETIME[2]` aren't equivalent.

Reply:
Yep, this could also be roundtripped properly. Should we do it in this PR?

Reply:
Up to you, but let's do it sooner rather than later if left for a followup PR, to reduce some of the tech debt.

Reply:
Added it in this PR